src/main.ts

```diff
@@ -38,27 +39,71 @@ async function run(): Promise<void> {
 
     // Download or build GraalVM
     let graalVMHome
-    switch (graalvmVersion) {
-      case c.VERSION_LATEST:
-        graalVMHome = await graalvm.setUpGraalVMLatest(gdsToken, javaVersion)
-        break
-      case c.VERSION_DEV:
-        graalVMHome = await graalvm.setUpGraalVMDevBuild(gdsToken, javaVersion)
-        break
-      default:
-        if (graalvmVersion.startsWith(c.MANDREL_NAMESPACE)) {
-          graalVMHome = await setUpMandrel(graalvmVersion, javaVersion)
-        } else {
-          if (enableCheckForUpdates) {
-            await checkForUpdates(graalvmVersion, javaVersion)
-          }
-          graalVMHome = await graalvm.setUpGraalVMRelease(
-            gdsToken,
-            graalvmVersion,
-            javaVersion
+    if (distribution.length > 0 || graalvmVersion.length == 0) {
+      switch (distribution) {
+        case c.DISTRIBUTION_GRAALVM:
+          graalVMHome = await graalvm.setUpGraalVMJDK(javaVersion)
+          break
+        case c.DISTRIBUTION_GRAALVM_COMMUNITY:
+          graalVMHome = await graalvm.setUpGraalVMJDKCE(javaVersion)
+          break
+        case c.DISTRIBUTION_MANDREL:
+          throw new Error(
+            `Mandrel requires the 'version' option (see https://github.com/graalvm/setup-graalvm/tree/main#options).`
           )
-        }
-        break
+        case '':
+          if (javaVersion === c.VERSION_DEV) {
+            core.info(
+              `This build is using the GraalVM Community Edition. To select a specific distribution, use the 'distribution' option (see https://github.com/graalvm/setup-graalvm/tree/main#options).`
+            )
+            graalVMHome = await graalvm.setUpGraalVMJDKDevBuild()
+          } else {
+            core.info(
+              `This build is using the new Oracle GraalVM. To select a specific distribution, use the 'distribution' option (see https://github.com/graalvm/setup-graalvm/tree/main#options).`
+            )
+            graalVMHome = await graalvm.setUpGraalVMJDK(javaVersion)
+          }
+        default:
+          throw new Error(`Unsupported distribution: ${distribution}`)
+      }
+    } else {
+      switch (graalvmVersion) {
+        case c.VERSION_LATEST:
+          if (javaVersion.startsWith('17') || javaVersion.startsWith('20')) {
```

Comment:
I wonder if we should use "javaVersion >= 20" using version comparison instead of `javaVersion.startsWith('20')`. Maybe we can do that only once we start to have `javaVersion` 21.

Reply:
Maybe we should do `|| semver.gte(javaVersion, 20)` for greater than? Note that this code path is only for `version: 'latest'` and that version is only needed for 22.3.X and earlier and currently Mandrel.

Reply:
Done as part of https://github.com/graalvm/setup-graalvm/pull/46/commits/be3e4f627247127382dfbd1eb9e90ab4e43f2dc7